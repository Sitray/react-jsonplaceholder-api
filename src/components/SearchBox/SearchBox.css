.search-bar{
    box-sizing: border-box;
    border: 1px solid black;
    width: 240px;
    height: 100%;
    padding: 0 12px;
    font-size: 14px;
    outline: none;
    text-overflow: ellipsis;
    border-radius: 7px;
}

nav {
    font-family:  Verdana, Geneva, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    border-radius: 3px;
}

nav ul {
    list-style: none;
    padding: 0;
    margin: 0;
    background: #333;
    border-radius: 7px;
}
nav li a {
    color: white;
    display: block;
    text-decoration: none;
    padding: .5em 1em;
}
nav li a:hover {
    background: tomato;
    color: white;
}

@media screen and (min-width: 480px) {
    nav ul {
      display: flex;
      justify-content: start;
    }
}
nav li{
    flex-grow:1;
  }