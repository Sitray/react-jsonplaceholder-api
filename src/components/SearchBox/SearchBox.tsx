/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';

import './SearchBox.css';

interface Props {
    placeholder: string,
    // eslint-disable-next-line no-unused-vars
    onSearch: (e: React.ChangeEvent<HTMLInputElement>) => void
}

const SearchBox = ({ onSearch, placeholder }: Props) => (
  <nav>
    <ul>
      <li><a href="#">Index</a></li>
      <li><a href="#">Nosotros</a></li>
      <li><a href="#">Servicios</a></li>
      <li><a href="#">Cont√°ctanos</a></li>
      <li>
        <input name="inputSearch" className="search-bar" type="text" placeholder={placeholder} onChange={onSearch} />
      </li>
    </ul>
  </nav>
);

export default SearchBox;
