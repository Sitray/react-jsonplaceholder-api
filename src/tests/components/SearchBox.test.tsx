/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import SearchBox from '../../components/SearchBox/SearchBox';

describe('Test on <SearchBox />', () => {
  const onSearch = jest.fn();
  const placeholder = 'Search';

  const wrapper = shallow(
    <SearchBox onSearch={onSearch} placeholder={placeholder} />,
  );

  test('should render correctly with props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
