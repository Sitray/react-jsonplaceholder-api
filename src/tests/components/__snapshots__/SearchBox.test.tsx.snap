// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Card /> should render correctly with props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Card
    body="This is a test"
    id={1}
    title="Test"
    userId={1}
  />
</ContextProvider>
`;

exports[`Test on <SearchBox /> should render correctly with props 1`] = `
<nav>
  <ul>
    <li>
      <a
        href="#"
      >
        Index
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Nosotros
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Servicios
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Cont√°ctanos
      </a>
    </li>
    <li>
      <input
        className="search-bar"
        name="inputSearch"
        onChange={[MockFunction]}
        placeholder="Search"
        type="text"
      />
    </li>
  </ul>
</nav>
`;
